### PR TITLE
[FAB-17140] update to use go chaincode

### PR DESCRIPTION
**Pull request by awjh-ibm**
Introduces a Commercial Paper Go contract to the existing Java and Node examples.

### CODE REVIEW

File: commercial-paper/README.md [hunk 1]
```diff
@@ -87,7 +87,15 @@ docker exec cliMagnetoCorp peer chaincode install -n papercontract -v 0 -p /opt/
 docker exec cliMagnetoCorp peer chaincode instantiate -n papercontract -v 0 -l java -c '{"Args":["org.papernet.commercialpaper:instantiate"]}' -C mychannel -P "AND ('Org1MSP.member')"
 ```
 
-> If you want to try both a Java and JavaScript Contract, then you will need to restart the infrastructure and deploy the other contract.
+**For a Go Contract:**
+
+```
+docker exec cliMagnetoCorp bash -c "cd /opt/gopath/src/github.com/hyperledger/fabric-samples/commercial-paper/organization/magnetocorp/contract-go GO111MODULE=on GOCACHE=on go mod vendor"
```
Comment (lehors) [inline commercial-paper/README.md]: This is missing a semi colon after the cd command, and GOCACHE=on didn't work for me - I got: build cache is required, but could not be located: GOCACHE is not an absolute path.
The following seems to work fine:
docker exec cliMagnetoCorp bash -c "cd /opt/gopath/src/github.com/hyperledger/fabric-samples/commercial-paper/organization/magnetocorp/contract-go; GO111MODULE=on go mod vendor"

File: commercial-paper/README.md [hunk 2]
```diff
@@ -87,7 +87,15 @@ docker exec cliMagnetoCorp peer chaincode install -n papercontract -v 0 -p /opt/
...
+docker exec cliMagnetoCorp peer chaincode install -n papercontract -v 0 -p github.com/hyperledger/fabric-samples/commercial-paper/organization/magnetocorp/contract-go -l golang
+docker exec cliMagnetoCorp peer chaincode instantiate -n papercontract -v 0 -l golang -c '{"Args":["org.papernet.commercialpaper:instantiate"]}' -C mychannel -P "AND ('Org1MSP.member')"
```
Comment (lehors) [inline commercial-paper/README.md]: This gave me:
Error: could not assemble transaction, err proposal response was not successful, error code 500, msg Channel 'mychannel' has been migrated to the new lifecycle, LSCC is now read-only
which made me realize that basic-network was upgraded to use the new cc lifecycle but commercial-paper (and maybe others) haven't. :-(

**Comment by mbwhite** [comment]
@lehors FYI I was looking at the Commercial Paper example on friday PM to check it worked - and also to confirm updates to the new lifecycle. :-)

**Comment by mbwhite** [comment]
@awjh-ibm if you rebase then the Javascript/Typescript tests will (should!) be ok.

File: commercial-paper/README.md (hunk 1 above)
Comment (awjh-ibm) [reply commercial-paper/README.md]: Thanks for pointing this out

File: commercial-paper/roles/magnetocorp.sh [hunk 3]
```diff
@@ -26,8 +26,13 @@ echo "
  docker exec cliMagnetoCorp peer chaincode install -n papercontract -v 0 -p /opt/gopath/src/github.com/contract-java -l java
  docker exec cliMagnetoCorp peer chaincode instantiate -n papercontract -v 0 -l java -c '{\"Args\":[\"org.papernet.commercialpaper:instantiate\"]}' -C mychannel -P \"AND ('Org1MSP.member')\"
 
+ Go Contract:
+ 
+ docker exec cliMagnetoCorp bash -c \"cd /opt/gopath/src/github.com/hyperledger/fabric-samples/commercial-paper/organization/magnetocorp/contract-go GO111MODULE=on GOCACHE=on go mod vendor\"
```
Comment (lehors) [inline commercial-paper/roles/magnetocorp.sh]: You need to fix that comment to match the updated README...

**Comment by lehors** [comment]
> @awjh-ibm if you rebase then the Javascript/Typescript tests will (should!) be ok.

Sorry, I don't understand this comment. What rebase are you talking about?

**Comment by lehors** [comment]
> > @awjh-ibm if you rebase then the Javascript/Typescript tests will (should!) be ok.
> 
> Sorry, I don't understand this comment. What rebase are you talking about?

oh! you're talking about the CI tests that kept failing, aren't you? Thank you!!!! :-)

**Comment by awjh-ibm** [comment]
Replaced by #102

File: commercial-paper/README.md (hunk 2 above)
Comment (trnhgquan) [reply commercial-paper/README.md:95]: @lehors  how did you fixed this issue? I got the same issue when init my new chaincode on test-network

Comment (lehors) [reply commercial-paper/README.md:95]: @trnhgquan These instructions have changed. When I use the latest README with Fabric 2.1 everything works (modulo some simple doc error I'm going to fix).

**Pull request closed without merging**